torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -147,12 +148,15 @@ Reducer::Reducer(
             at::zeros({static_cast<long>(variable_count)}, options);
       }
 
-      // This tensor needs to be on the same device as replica because backend
-      // such as NCCL may not support CPU tensors, and hence it might not work
-      // if we always put it on CPU.
-      options = options.device(replicas_[i][0].device());
-      local_used_maps_dev_[i] =
-          at::empty({static_cast<long>(variable_count)}, options);
+      if (find_unused_parameters_) {
+        local_used_maps_dev_.resize(replica_count);
```

Comment:
should the resize happen before the for loop? Is there any reason that we need this in every iteration?

Reply:
> should the resize happen before the for loop? Is there any reason that we need this in every iteration?

Right, I will fix it.